pkg/controller/cluster/cluster_controller.go

```diff
@@ -201,15 +224,20 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, nil
 
 	} else if utils.Contains(instance.ObjectMeta.Finalizers, clusterFinalizer) {
-		// our finalizer is present, so delete cluster first
-		err := k.DeleteCluster(instance.Spec.KopsConfig)
+
+		//check if cluster still exists
+		err := k.GetCluster(instance.Spec.KopsConfig)
 		if err != nil {
-			// FIXME - Ensure that delete implementation is idempotent and safe to invoke multiple times.
-			// If we call delete and the cluster is not present it will cause error and it will keep erroring out
-			//return reconcile.Result{}, err
+			reqLogger.Info("Cluster is already deleted...")
```

Comment:
Can we not check explicitly check if cluster is present or not (instead of err != nil)?

Reply:
@hryan-ib 
I think that is fair, we could change k.GetCluster() to return (bool, error)
Then the code would be: 
found, err := k.GetCluser()

Reply:
Include the error in the log message

Reply:
I have updated the code to use a boolean. It will check that the error is for not finding the cluster ("exit status 1") and will report that as a cluster not found. Any other error will be treated as a command error and will exit reconcile. If cluster found, the boolean is true and it will continue reconciling